### DISCUSSION TITLE
如何自定义编译的.so的名字

### DISCUSSION THREAD

**Discussion by ZiJiaW** [xmake-io/xmake#2188] [Q&A] [answered]
默认情况下xmake编译
```
target("name")
    set_kind("shared")
    ....
```
会生成名为libname.so的库文件，请问能否指定生成的库文件的名字，比如我想要它输出name.so，like gcc ... -o name.so

**Comment by waruqi** [comment, marked as answer]
set_filename("name") or set_prefixname("")
